### DISCUSSION TITLE
Ghostty shows pwd/lock indicator when using Chafa, is it a Chafa issue?

### DISCUSSION THREAD

**Discussion by edrozenberg** [hpjansson/chafa#236] [Q&A] [answered]
Chafa version 1.14.5 (Homebrew)
Ghostty 1.0.0
MacOS 15.2 (24C101)

When using Chafa with the Ghostty terminal app, Ghostty shows the lock icon and security info that it normally shows at a password prompt or at secure text entry.

From the Ghostty developer:
[mitchellh]
> This is based on the heuristic we use to detect password input. I don't think there's a way for us to fix this because that's the same heuristic used by iTerm and others to enable this feature. I would recommend reporting to Chafa to try to ease up on their ioctls here.

https://github.com/user-attachments/assets/1299907d-1415-439e-9ec6-8e76d4d5b870

**Comment by hpjansson** [comment, marked as answer]
Hi, thanks for bringing this up. It should probably be filed as an issue, though. Do you know which ioctl mitchellh is referring to? Does passing `--polite on` help?

Edit: When `tcsetattr()` is used to turn off echo, ghostty interprets it to mean there's a password prompt. You can indeed work around it by passing `--polite on`. I'll change it to emit a control sequence instead, which does not trigger the heuristic.

**Reply by edrozenberg** [reply]
That's great, `--polite on` is working well to suppress the password issue. Thank you!
